tests/callbacks/test_callbacks.py

```diff
@@ -137,6 +138,41 @@ def test_resume_callback_state_saved_by_type(tmpdir):
     assert callback.state == 111
 
 
+class OldStatefulCallbackHooks(Callback):
+    def __init__(self, state):
+        self.state = state
+
+    @property
+    def state_key(self):
+        return type(self)
+
+    def on_save_checkpoint(self, trainer, pl_module, checkpoint):
+        return {"state": self.state}
+
+    def on_load_checkpoint(self, trainer, pl_module, callback_state):
+        self.state = callback_state["state"]
+
+
+def test_resume_callback_state_saved_by_type_hooks(tmpdir):
+    """Test that a legacy checkpoint that didn't use a state key before can still be loaded, using
+    on_save/load_checkpoint."""
+    # TODO: remove on_save_checkpoint() -> dict
+    # and on_load_checkpoint(callback_state) support in v1.8
```

Comment:
Once again I'm struggling to understand why we have a todo to remove the hook. 
I thought we proposed to standardize the `on_*_checkpoint` methods to take the full checkpoint dict as an argument and remove the return value.

The todo should say that the test gets removed?

Reply:
todo is only to remove on_save_checkpoint **-> dict** and on_load_checkpoint(**callback_state**) support.
will add a line clarifying "in favor of on_save_checkpoint **-> None** and on_load_checkpoint(**checkpoint**)"